pypeit/scripts/show_2dspec.py

```diff
@@ -137,7 +141,7 @@ def main(args):
         # Try to read the Spec2DObj using the current datamodel, but allowing
         # for the datamodel version to be different
         try:
-            spec2DObj = spec2dobj.Spec2DObj.from_file(args.file, detname, chk_version=False)
+            spec2DObj = spec2dobj.Spec2DObj.from_file(args.file, detname, chk_version=chk_version)
```

Comment:
Thanks for the prompt to think this through more deeply.  I think we should leave it as is for the following reason.

The `chk_version` approach is very basic.  It just checks if the version numbers are different, and raises a `PypeItDataModelError` if they aren't.  But the datamodel code may throw a `PypeItDataModelError` for other reasons.  So it's possible for their to be cases when someone tries `chk_version=False`, but this try block still catches an exception.  In that case, we still want the code to try to pull the necessary data from the spec2d file and show it.  The warning should hopefully prompt the user to appreciate that, if the results don't look right, it could be because the datamodel changed and they should re-reduce the data.

What do you think?

Reply:
Hi @kbwestfall, in general I'm OK to keep try/except and, as you said, it makes sense for when  `chk_version=False`. What I am most worried is when `chk_version=True`, which is the default. If users run `pypeit_show_2dspec` without `--try_old` (as they are used to) with an older version of spec2d, the script still shows something (because of the try/except and the code below that try to pull the necessary data from the spec2d file and show it) and I'm assuming users will not read the warning and will be OK with what they get. The problem is that the code after `if spec2DObj is None:` is not often updated, so it may not show the correct frames (I tried it with a v1.13.1 spec2d and the SCIIMG is not right).
That said, I think the ideal situation would be that if the script is run with `chk_version=True`, the script faults and directs the user to use `chk_version=False`. If the script is run with `chk_version=False`, try/except caches any `PypeItDataModelError` and allows the script to show something anyway using the code after `if spec2DObj is None:`.

Reply:
Ah, I get it now.  Yes, I like your suggested ideal situation.  I've pushed changes that should implement this.  Can you give it a test run?

Reply:
Thanks @kbwestfall! This looks great!